sdk/spring/azure-spring-integration-servicebus/src/main/java/com/azure/spring/integration/servicebus/converter/ServiceBusMessageConverter.java

```diff
@@ -124,13 +142,21 @@ protected void setCustomHeaders(MessageHeaders headers, ServiceBusMessage messag
         setValueIfHasText(headers, REPLY_TO_SESSION_ID, message.getReplyToSessionId());
         setValueIfHasText(headers, SESSION_ID, message.getSessionId());
 
-        headers.putAll(message.getApplicationProperties());
+        message.getApplicationProperties().forEach((key, value) -> {
+            if (!headers.containsKey(key)) {
+                headers.put(key, value);
+            }
+        });
 
         return Collections.unmodifiableMap(headers);
     }
 
-    private Optional<String> getStringHeader(MessageHeaders springMessageHeaders, String key) {
-        return Optional.ofNullable(springMessageHeaders.get(key)).map(Object::toString).filter(StringUtils::hasText);
+    private Optional<String> getStringHeader(MessageHeaders springMessageHeaders, Set<String> copyHeaders, String key) {
```

Comment:
can you explain a little bit of what this method is doing in javadoc? and stop using this optional way since it really makes simple things complex

Reply:
Javadoc added. I roll back the method of getStringHeader for its original correct usage of Optional.